nomad/node_endpoint.go

```diff
@@ -820,10 +820,50 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 		return fmt.Errorf("must update at least one allocation")
 	}
 
+	// Ensure that evals aren't set from client RPCs
+	// We create them here before the raft update
+	if len(args.Evals) != 0 {
+		return fmt.Errorf("evals field must not be set ")
+	}
+
 	// Update modified timestamp for client initiated allocation updates
-	now := time.Now().UTC().UnixNano()
+	now := time.Now()
+	var evals []*structs.Evaluation
+
 	for _, alloc := range args.Alloc {
-		alloc.ModifyTime = now
+		alloc.ModifyTime = now.UTC().UnixNano()
+
+		// Add an evaluation if this is a failed alloc that is eligible for rescheduling
+		if alloc.ClientStatus == structs.AllocClientStatusFailed {
+			ws := memdb.NewWatchSet()
+			job, err := n.srv.State().JobByID(ws, alloc.Namespace, alloc.JobID)
+			if err != nil {
+				n.srv.logger.Printf("[ERR] nomad.client: Unable to find jobid %v", alloc.JobID)
+				return err
+			}
+			if job == nil {
+				return fmt.Errorf("[ERR] nomad.client: Unable to find jobid %v", alloc.JobID)
```

Comment:
The format of `[ERR] nomad.client:` is for logging only not for returning errors